tests/selectors_test.py

```diff
@@ -85,22 +82,140 @@ def test_categorical(
+    ):
+        request.applymarker(pytest.mark.xfail)
+
+    ts1 = datetime(2000, 11, 20, 18, 12, 16, 600000)
+    ts2 = datetime(2020, 10, 30, 10, 20, 25, 123000)
+
+    data = {
+        "numeric": [3.14, 6.28],
+        "ts": [ts1, ts2],
+    }
+    time_units: list[Literal["ns", "us", "ms", "s"]] = ["ms", "us", "ns"]
+
+    df = nw.from_native(constructor(data)).select(
+        nw.col("numeric"),
+        *[
+            nw.col("ts").cast(nw.Datetime(time_unit=tu)).alias(f"ts_{tu}")
+            for tu in time_units
+        ],
+        *[
+            nw.col("ts")
+            .dt.replace_time_zone("Europe/Lisbon")
+            .cast(nw.Datetime(time_zone="Europe/Lisbon", time_unit=tu))
+            .alias(f"ts_lisbon_{tu}")
+            for tu in time_units
+        ],
+        *[
+            nw.col("ts")
+            .dt.replace_time_zone("Europe/Berlin")
+            .cast(nw.Datetime(time_zone="Europe/Berlin", time_unit=tu))
+            .alias(f"ts_berlin_{tu}")
+            for tu in time_units
+        ],
+    )
+
+    assert df.select(ncs.datetime()).collect_schema().names() == [
+        "ts_ms",
+        "ts_us",
+        "ts_ns",
+        "ts_lisbon_ms",
+        "ts_lisbon_us",
+        "ts_lisbon_ns",
+        "ts_berlin_ms",
+        "ts_berlin_us",
+        "ts_berlin_ns",
+    ]
+    assert df.select(ncs.datetime(time_unit="ms")).collect_schema().names() == [
+        "ts_ms",
+        "ts_lisbon_ms",
+        "ts_berlin_ms",
+    ]
+    assert df.select(ncs.datetime(time_unit=["us", "ns"])).collect_schema().names() == [
+        "ts_us",
+        "ts_ns",
+        "ts_lisbon_us",
+        "ts_lisbon_ns",
+        "ts_berlin_us",
+        "ts_berlin_ns",
+    ]
+
+    assert df.select(ncs.datetime(time_zone=None)).collect_schema().names() == [
+        "ts_ms",
+        "ts_us",
+        "ts_ns",
+    ]
+    assert df.select(ncs.datetime(time_zone="*")).collect_schema().names() == [
+        "ts_lisbon_ms",
+        "ts_lisbon_us",
+        "ts_lisbon_ns",
+        "ts_berlin_ms",
+        "ts_berlin_us",
+        "ts_berlin_ns",
+    ]
+    assert df.select(
+        ncs.datetime(time_zone=[None, "Europe/Berlin"])
+    ).collect_schema().names() == [
+        "ts_ms",
+        "ts_us",
+        "ts_ns",
+        "ts_berlin_ms",
+        "ts_berlin_us",
+        "ts_berlin_ns",
+    ]
+
+    assert df.select(
+        ncs.datetime(time_unit="ns", time_zone=[None, "Europe/Berlin"])
+    ).collect_schema().names() == ["ts_ns", "ts_berlin_ns"]
+
+    assert df.with_columns(
+        nw.col("ts_ms").dt.replace_time_zone("UTC").alias("ts_utc")
+    ).select(
+        ncs.datetime(time_unit=["ms", "us"], time_zone=[None, timezone.utc])
+    ).collect_schema().names() == ["ts_ms", "ts_us", "ts_utc"]
+
+
+def test_datetime_spark_and_duckdb(constructor: Constructor) -> None:
+    if not ("pyspark" in str(constructor) or "duckdb" in str(constructor)):
+        pytest.skip()
```

Comment:
these should be supported now?

Reply:
This is an "easy"-er example test for duckdb and pyspark since we don't support converting with timezones for those backends

Reply:
Meaning, all other backends will pass this test anyway, while duckdb and pyspark won't pass the other one. No harm in having all backends here